lib/core.js

```diff
@@ -407,7 +407,7 @@ declare class Promise<R> {
 
     then<U>(
       onFulfill?: (value: R) => Promise<U> | U,
-      onReject?: (error: any) => Promise<U> | U
+      onReject?: (error: any) => ?Promise<U> | U
```

Comment:
Interesting. I think `Promise<U> | U | void` would be more precise here. `?Promise<U> | U` also allows `null`.